juniper_official/routing/bfd-session-state.rule

```diff
@@ -0,0 +1,134 @@
+                iAgent {
+                    file bfd-session-state.yml;
+                    table BfdSessionTable;
+                    frequency 180s;
+                }
+            }
+            field session-neighbor {
+                sensor bfd-sensor {
+                    where "session-neighbor =~ /{{bfd-session-id}}/";
+                    path session-neighbor;
+                }
+                type string;
+                description "BFD neighbor ip";
+            }
+            field session-state {
+                sensor bfd-sensor {
+                    path session-state;
+                }
+                type string;
+                description "BFD session state";
+            }
+            field remote-state {
+                sensor bfd-sensor {
+                    path remote-state;
+                }
+                type string;
+                description "BFD remote state";
+            }
+            trigger bfd-session-state {
+                synopsis "BFD neighbor state kpi";
+                description "Sets health based on BFD neighbor state.";
+                frequency 1offset;
+                term is-bfd-session-down {
+                    when {
+                        does-not-match-with "$session-state" UP {
+                            ignore-case;
+                        }
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "BFD session to $session-neighbor is $session-state. Remote state is $remote-state";
+                        }
+                    }
+                }
+                term bfd-session-up {
+                    then {
+                        status {
+                            color green;
+                            message "BFD session to $session-neighbor is $session-state. Remote state is $remote-state";
+                        }
+                    }
+                }
+            }
+            variable bfd-session-id {
+                value .*;
+                description "BFD session IP to monitor, regular expression, e.g. '10.10.10.1|10.20.20.1|172.16.1.1'";
+                type string;
+            }
+            rule-properties {
+                version 2;
+                contributor juniper;
+                category advanced;
+                is-scaling-rule {
+                    description "Fields:session-neighbor ; Directly impacted by number of bfd sessions running in each network device";
+                }				
+                supported-healthbot-version 1.0.1;
+                supported-devices {
+                    juniper {
+                        operating-system junos {
+                            products EX {
+                                platforms All {
```

Comment:
there is nothing like platforms ALL, this was a mistake in older rules, either mention each of the products or put a default sensor at platform level